apps/src/lab2/responseValidators.ts

```diff
@@ -1,32 +1,49 @@
 import {ResponseValidator} from '@cdo/apps/util/HttpClient';
-import {BlocklySource, LevelProperties, ProjectSources} from './types';
+import {AppName, BlocklySource, LevelProperties, ProjectSources} from './types';
+import Lab2Registry from './Lab2Registry';
+import {BLOCKLY_LABS} from './constants';
+import {NestedSourceCode} from '@cdo/apps/pythonlab/pythonlabRedux';
 
-export const SourceResponseValidator: ResponseValidator<
+// Validator for Blockly sources.
+export const BlocklySourceResponseValidator: ResponseValidator<
   ProjectSources
 > = response => {
-  if (!response.source) {
-    throw new ValidationError('Missing required field: source');
-  }
+  const blocklyValidator = (source: string) => {
+    const blocklySource = parseJSON<BlocklySource>(source);
+    if (blocklySource.blocks === undefined) {
+      throwMissingFieldError('blocks');
+    }
+  };
 
-  // Currently only Blockly JSON sources are supported.
-  let blocklySource;
-  try {
-    blocklySource = JSON.parse(response.source as string) as BlocklySource;
-  } catch (e) {
-    throw new ValidationError('Error parsing JSON: ' + e);
-  }
-  if (blocklySource.blocks === undefined) {
-    throw new ValidationError('Missing required field: blocks');
-  }
+  return sourceValidatorHelper(response, blocklyValidator);
+};
 
-  return response as unknown as ProjectSources;
+// Validator for Python sources.
+export const PythonSourceResponseValidator: ResponseValidator<
+  ProjectSources
+> = response => {
+  const pythonValidator = (source: string) => {
+    const pythonSource = parseJSON<NestedSourceCode>(source);
```

Comment:
To confirm - is python source code stored as stringified JSON in the main.json file? Wondering because don't we store Java lab code as plain non-stringified JSON, and if so would we want to do the same for multi-file python projects?

Reply:
very good point. I was reproducing the blockly pattern of stringified json but using the Java Lab pattern is better for multi file. I'll update.

Reply:
I refactored this to match the Java Lab pattern!